### DISCUSSION TITLE
PythonEngine fails to initialize in VSIX extension

### DISCUSSION THREAD

**Discussion by cpgames** [pythonnet/pythonnet#2405] [Q&A] [answered]
### Environment

-   Pythonnet version: 3.0.3, 3.1.0
-   Python version: 3.09
-   Operating System: Win11
-   .NET Runtime: 4.8

### Details

-   Describe what you were trying to get done.

I am trying to use pythonnet package in a VSIX Visual Studio extension.
The Experimental VS Instance initializes pythonnet fine, however deploying an extension causes this crash.
I suspect there may be some threading issues, but am not sure what to look for.

-   What commands did you run to trigger this issue? If you can provide a
    [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
    this will help us understand the issue.

```
try
{
    Runtime.PythonDLL = pythonFile.FullName;
    PythonEngine.PythonHome = pythonFile.DirectoryName!;
    OutputWindowHelper.LogInfo("PythonEngine", "Initializing Python engine.");
    PythonEngine.Initialize(); // crash here
    PythonEngine.BeginAllowThreads();
}
catch (Exception ex)
{
    OutputWindowHelper.LogError(ex);
}
```

-   If there was a crash, please include the traceback here.
```
[INFO] - PythonEngine: Initializing Python engine.
[ERROR]: Object reference not set to an instance of an object.
[TRACE]:    at Python.Runtime.ClassManager.GetClassInfo(Type type, ClassBase impl)
   at Python.Runtime.ClassManager.InitClassBase(Type type, ClassBase impl, ReflectedClrType pyType)
   at Python.Runtime.ReflectedClrType.GetOrCreate(Type type)
   at Python.Runtime.ModuleObject.GetAttribute(String name, Boolean guess)
   at Python.Runtime.ModuleObject.LoadNames()
   at Python.Runtime.ImportHook.UpdateCLRModuleDict()
   at Python.Runtime.PythonEngine.Initialize(IEnumerable`1 args, Boolean setSysArgv, Boolean initSigs)
   at Python.Runtime.PythonEngine.Initialize()
   at cpGames.VSA.ToolAPI.<InitializePythonEngineAsync>d__2.MoveNext() in ToolAPI.cs:line 69 // PythonEngine.Initialize(); call
```

**Comment by cpgames** [comment, marked as answer]
Why you close my issue?

Here's your fix:

```
internal static bool ShouldBindEvent(EventInfo ei)
        {
            var mb = ei.GetAddMethod(true);
            if (mb == null)
            {
                return false;
            }
            return ShouldBindMethod(mb);
        }
```

Check your nulls, thank me later.

**Comment by adrctr** [comment]
Hello ,

I have the same issue, I am trying to use pythonnet package in a VSIX Visual Studio extension.

![image](https://github.com/pythonnet/pythonnet/assets/27783562/dbd88d82-7b22-4472-8f51-d3a5124e6ee7)

I am going to apply this fix above made by @cpgames and see if the problem persists.

Do you intend to release this fix in a future version of pythonnet ? 

Regards,
